src/modeler/AutoRest.Swagger/Model/ServiceDefinition.cs

```diff
@@ -102,6 +103,7 @@ public ServiceDefinition()
         [Rule(typeof(ProvidersPathValidation))]
         [CollectionRule(typeof(XmsPathsMustOverloadPaths))]
         [CollectionRule(typeof(BodyPropertiesNamesCamelCase))]
+        [Rule(typeof(TrackedResourceListByImmediateParent))]
```

Comment:
I have reservations about applying this to custom paths, they overload an existing path, and may return non-resource items, we need to confirm this.